architecture/ambient/lifecycle.md

```diff
@@ -0,0 +1,136 @@
+When this occurs, the plugin will call out to the CNI Agent to program the network.
+This includes setting up networking rules both within the pod network namespace, and also on the host network.
+
+An alternative flow is when a pod is enabled after startup.
+In this case, the CNI Agent is watching for Pod events from the API server directly and performing the same setup.
+
+Once the network is configured, the CNI Agent will signal to Ztunnel to start running within the Pod.
+This is done by the [ZDS](../../pkg/zdsapi/zds.proto) API.
+This will send some identifying information about the Pod to Ztunnel, and, importantly, the Pod's network namespace file descriptor.
+
+Ztunnel will use this to enter the Pod network namespace and start various listeners (inbound, outbound, etc).
+
+> [!NOTE]
+> While Ztunnel runs as a single shared binary on the node, each individual pod gets its own unique set of listeners within its own network namespace.
+
+## Pod Startup
+
+### Pod Startup Requirements
+
+During Pod startup, we MUST meet the constraint that the network is ready the instant the application starts (including `initContainers`!).
+This is critical -- issues around this were the [top issue with sidecars](https://github.com/istio/istio/issues/11130).
+
+### Pod Startup Implementation
+
+The key mechanism to maintain this property is the CNI plugin.
+The CNI plugin synchronously runs before application containers start, ensure we can setup the network configuration and start Ztunnel.
+Only once those two conditions are met do we mark the CNI plugin as successfully complete.
+
+From the above we are ensured that when the application starts and opens a connection, Ztunnel will be ready to `accept()` it.
+However, this is not enough to fully serve the request - Ztunnel will need the certificate (from the CA) and full workload information (from the XDS server).
+This is done through a waiting mechanism: Ztunnel will accept all connections, but it will not start forwarding traffic until all the above is fetched.
+This "lazy loading" optimizes pod startup time, for the common case where an application is not sending outbound traffic instantly.
+For applications that do, they will see a slight latency increase during startup; most applications can handle this far better than failed connections.
+
+## Pod Shutdown
+
+### Pod Shutdown Requirements
+
+During Pod shutdown, we MUST meet the constraint that the Pod shutting down can send traffic up until its full stopped.
+Pod shutdown can be a long process; while `terminationGracePeriodSeconds` is typically ~30s, it can be extremely high, leaving terminating pods running for extended periods of time with an expectation of networking.
+
+Additionally, we MUST accept incoming traffic during this time, both new and existing.
+Typically, it is rare to get new connections after shutdown has started - once shutdown is triggered the endpoint is marked `NotReady` and removed from load balancing consideration.
```

Comment:
For example using persistent sessions or pod-to-pod communication will keep sending new connections after shutdown has started.